proxy/src/schema.rs

```diff
@@ -44,24 +77,122 @@ impl Into<ProjectId> for IdInput {
     }
 }
 
-/// Encapsulates read paths in API.
-pub struct Query;
+/// Object representing a git branch.
+#[derive(GraphQLObject)]
+struct Branch {
+    /// Name of the branch.
+    name: String,
+}
 
-#[juniper::object(Context = Context)]
-impl Query {
-    fn apiVersion() -> &str {
-        "1.0"
-    }
+/// Object representing a git tag.
+///
+/// We still need full tag support.
+#[derive(GraphQLObject)]
+struct Tag {
+    /// Name of the tag.
+    name: String,
+}
 
-    fn projects(ctx: &Context) -> FieldResult<Vec<Project>> {
-        Ok(ctx.source.get_all_projects())
-    }
+/// Representation of a person (e.g. committer, author, signer) from a repository. Usually
+/// extracted from a signature.
+#[derive(GraphQLObject)]
+struct Person {
+    /// Name part of the commit signature commit.
+    name: String,
+    /// Email part of the commit signature commit.
+    email: String,
+    /// Reference (url/uri) to a persons avatar image.
+    avatar: String,
+}
 
-    fn project(ctx: &Context, id: IdInput) -> FieldResult<Option<Project>> {
-        Ok(ctx.source.get_project(id.into()))
+/// Representation of a code commit.
+#[derive(GraphQLObject)]
+struct Commit {
+    /// Identifier of the commit in the form of a sha1 hash. Often referred to as oid or object id.
+    sha1: String,
+    /// The author of the commit.
+    author: Person,
+    /// The summary of the commit message body.
+    summary: String,
+    /// The entire commit message body.
+    message: String,
+    /// The recorded time of the commit.
+    time: String,
```

Comment:
I think we should be a bit more explicit whether this is the [author or commit time](https://stackoverflow.com/questions/11856983/why-git-authordate-is-different-from-commitdate#11857467). I also noted it in the [schema](https://github.com/radicle-dev/radicle-upstream/blob/xla/surf-integration/app/schema.graphql#L13).

Reply:
We need both you are right. There are actualy organised under two fields `author` and `commiter` which are what's known as a `signature`.

So we can have two fields on the commit `authorDate` and `committerDate` or we expand to proper signature data for each. What you think?

Reply:
I'm happy with 2 extra fields on the commit.

I feel like the term "signature" is a bit overloaded already (think commit signing), so if you prefer on extracting the fields, I'd urge to choose a different name for it.

Reply:
The signature here is referring to the "commit signing" one from each party an author signature and a commiter signature, they refer to the same concept. :>

Reply:
I see, in that case, extracting it probably makes sense.

Reply:
I went for the intermediate solution and ultimately what you proposed to begin with. The field is now explcitiley stating that it is the committer time. This is a shorthand until we expose `author` and `comitter` signatures propr. Check 0bb3a34 for the actual change.